tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/eval/ParseEvaluation.scala

```diff
@@ -0,0 +1,148 @@
+    goldParseBank: ParseBank
+  ): Unit = {
+
+    val goldScoringFunctions: Seq[ParseScore] = Seq(
+      UnlabeledAttachmentScore(goldParseBank),
+      LabeledAttachmentScore(goldParseBank),
+      UnlabeledLostTokens(goldParseBank),
+      LabeledLostTokens(goldParseBank),
+      PostagAccuracy(goldParseBank)
+    )
+    val allScores = for {
+      scoringFunction <- goldScoringFunctions
+    } yield {
+      (scoringFunction.name, ParseEvaluation.scoreParseSource(scoringFunction, candidateSource))
+    }
+    allScores foreach {
+      case (scoreName, (userNumer, userDenom)) =>
+        println(f"$scoreName: ${100 * userNumer.toDouble / userDenom}%.2f" + "%")
+    }
+  }
+
+  /** Runs a standard set of analyzers on a set of candidate parses with respect to a
+    * bank of gold parses.
+    *
+    * @param candidateSource the source of candidate parses
+    * @param goldParseBank a bank containing the gold parses
+    */
+  def performStandardAnalysis(
+    candidateSource: PolytreeParseSource,
+    goldParseBank: ParseBank
+  ): Unit = {
+
+    val analyzers: Seq[ParseAnalyzer] = Seq(
+      MisattachmentAnalyzer(goldParseBank, ignoreLabel = true),
+      LostTokensAnalyzer(goldParseBank),
+      CposErrorAnalyzer(goldParseBank)
+    )
+    val allAnalyses = for {
+      analyzer <- analyzers
+    } yield {
+      (analyzer.name, ParseEvaluation.analyzeParseSource(analyzer, candidateSource))
+    }
+    allAnalyses foreach {
+      case (analyzerName, analysis) =>
+        println(s"ARCLABEL\t$analyzerName")
+        analysis foreach {
+          case (label, count) =>
+            println(s"${label.toLowerCase}\t$count")
+        }
+    }
+  }
+
+  /** Runs a standard contrastive evaluation of a set of candidate parsers ("users")
+    * with respect to a bank of gold parses.
+    *
+    * @param candidateSources the map from user names to candidate parses
+    * @param goldParseBank a bank containing the gold parses
+    */
+  def performStandardMultiUserEvaluation(
+    candidateSources: Map[String, PolytreeParseSource],
+    goldParseBank: ParseBank
+  ): Unit = {
+
+    val goldScoringFunctions: Seq[ParseScore] = Seq(
+      UnlabeledAttachmentScore(goldParseBank),
+      LabeledAttachmentScore(goldParseBank),
+      UnlabeledLostTokens(goldParseBank),
+      LabeledLostTokens(goldParseBank),
+      PostagAccuracy(goldParseBank)
+    )
+    val allScores = for {
+      scoringFunction <- goldScoringFunctions
+    } yield {
+      (scoringFunction.name, for {
+        (userIName, userIParses) <- candidateSources.toSeq
+      } yield {
+        val score = ParseEvaluation.scoreParseSource(scoringFunction, userIParses)
+        (userIName, score)
+      })
+    }
+    allScores foreach {
+      case (scoreName, userScores) =>
+        println(s"USER\t$scoreName")
+        userScores foreach {
+          case (username, (numer, denom)) =>
+            println(f"$username\t${numer.toDouble / denom}%.3f")
```

Comment:
Agreed. I'll probably defer that to the next PR (this code is still being refined as I continue the user study analysis).